test/Mono.Linker.Tests.Cases/Logging/CommonLogs.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Logging {
+
+#if !NETCOREAPP
+	[IgnoreTestCase ("Can be enabled once MonoBuild produces a dll from which we can grab the types in the Mono.Linker namespace.")]
+#else
+	[SetupCompileBefore ("LogStep.dll", new [] { "Dependencies/LogStep.cs" }, new [] { "illink.dll" })]
```

Comment:
I do not believe that referencing the linker from a test case is a good path to go down.  The test cases were intended to test controlled use cases.  linking the linker itself is not a controlled test case and is bound to result in tests that are high maintenance.

If you want to write a test like this I think it should live in `Mono.Linker.Tests` as a normal unit test.  Maybe something similar to `IndividualTests` if you still want to use a test case and the testing infrastructure to drive running of the linker.

Reply:
Thinking about it more...I don't think using the test case infrastructure in any way is the way to go here.

It looks more like you want to write a unit test that sets up a `LinkContext` and then calls
```
				MessageOrigin msgOrigin = new MessageOrigin ("logtest");
			Message msgError = new Message (msgOrigin, MessageCategory.Error, MessageCode.L100, text: "Error");
			Message msgWarning = new Message (msgOrigin, MessageCategory.Warning, MessageCode.L400, text: "Warning");
			Message msgInfo = new Message (new MessageOrigin(), MessageCategory.Info, MessageCode.L900, text: "Info");
			context.LogMessage (msgError);
			context.LogMessage (msgWarning);
			context.LogMessage (msgInfo);
```

And then you assert what is in the `ILogger` class that you setup with the `LinkContext`.

If you wanted to test logging end to end via a test case then I would setup a scenario that cases a message to be logged and then assert that message is in the log.  If you don't have messages hooked up yet, then I would hold off on a test case until you do.

Reply:
I'll move the test to `Mono.Linker.Tests` for now. I agree that a better way to test this would be to have a scenario that triggers a warning/error to be logged, but in this PR I'm just adding the basic infrastructure for logging support through MSBuild. I'll do as you advice and write tests that trigger the warnings once I update all the places where we log warnings/errors in the linker to use this.